src/main/kotlin/ar/edu/unahur/obj2/servidorWeb/Analizador.kt

```diff
@@ -0,0 +1,77 @@
+package ar.edu.unahur.obj2.servidorWeb
+
+import java.time.LocalDateTime
+
+abstract class Analizador() {
+    abstract fun analizar(respuesta: Respuesta)
+}
+
+object analizadorDeRespuestas : Analizador(){
+
+    val demoraMinima = 50
+    val respuestasDemoradas = mutableListOf<Respuesta>()
+
+    override fun analizar(respuesta: Respuesta) {
+        if (respuesta.tiempo > this.demoraMinima){
+            respuestasDemoradas.add(respuesta)
+        }
+    }
+
+    fun respuestasDemoradasModulo(modulo: Modulo): Int {
+        return respuestasDemoradas.filter { respuesta -> respuesta.body == modulo.body }.size
+    }
+}
+
+object analizadorDeIp : Analizador() {
+
+    val ipsSospechosas = mutableListOf<String>()
+    val respuestasConIpSospechosa = mutableListOf<Respuesta>()
+
+    fun agregarIpSospechosa(ipSospechosa: String){
+        ipsSospechosas.add(ipSospechosa)
+    }
+
+    override fun analizar(respuesta: Respuesta) {
+        if (ipsSospechosas.contains(respuesta.pedido.ip)){
+            respuestasConIpSospechosa.add(respuesta)
+        }
+    }
+
+
+    fun cuantosPedidosSeRealizaronConIp(ipSospechosa: String): Int{
+        return respuestasConIpSospechosa.filter { respuesta -> respuesta.pedido.ip == ipSospechosa }.size
+    }
+
+    fun cualFueElModuloMasConsultado(){
+        // Intente varias cosas pero no lo pude sacar
+    }
+
+    fun ipsQueRequirieronRuta(Ruta: String): Set<String> {
+        return respuestasConIpSospechosa.filter{respuesta -> respuesta.pedido.url.contains(Ruta)}.map { respuesta -> respuesta.pedido.ip }.toSet()
+    }
+}
+
+object analizadorEstadisticas : Analizador() {
+
+    val respuestasParaEstadisticas = mutableListOf<Respuesta>()
+
+    override fun analizar(respuesta: Respuesta) {
+        respuestasParaEstadisticas.add(respuesta)
+    }
+
+    fun tiempoRespuestaPromedio(): Int {
+        return respuestasParaEstadisticas.sumBy { respuesta -> respuesta.tiempo } / respuestasParaEstadisticas.size
```

Comment:
Está muy bien, otra forma de hacerlo:
```suggestion
        return respuestasParaEstadisticas.map { respuesta -> respuesta.tiempo }.average()
```